device/common/include/traccc/clusterization/device/impl/aggregate_cluster.ipp

```diff
@@ -67,14 +93,26 @@ inline void aggregate_cluster(
 
             if (weight > this_module.threshold) {
                 totalWeight += weight;
-                const point2 cell_position =
+                scalar weight_factor = weight / totalWeight;
+
+                point2 cell_position =
                     traccc::details::position_from_cell(this_cell, this_module);
-                const point2 prev = mean;
-                const point2 diff = cell_position - prev;
 
-                mean = prev + (weight / totalWeight) * diff;
+                if (!first_processed) {
+                    offset = cell_position;
+                    first_processed = true;
+                }
+
+                cell_position = cell_position - offset;
+
+                const point2 diff_old = cell_position - mean;
+                mean = mean + diff_old * weight_factor;
+                const point2 diff_new = cell_position - mean;
+
+#pragma unroll
```

Comment:
You should of course use the macro introduced for Alpaka. But I wonder... is it even worth to do all this @#$^ for this tiny/trivial loop? Why not just write out the two statements with a copy-paste and be done with it? :thinking:

Reply:
What I wonder is: Vectorization is not going to be a thing here, right? (Or could it be?) If not, just writing 2 statements instead of the for-loop would seem to be the way to go.

Reply:
I think the nicest solution would be to use the `point2` semantics, but I don't think it supports element-wise multiplication. :slightly_frowning_face: